rollup.config.js

```diff
@@ -1,97 +1,76 @@
-'use strict';
+import babel from '@rollup/plugin-babel';
+const commonjs = require('@rollup/plugin-commonjs');
+const { nodeResolve } = require('@rollup/plugin-node-resolve');
+const nodePolyfills = require('rollup-plugin-node-polyfills');
 
-const pkg = require('./package.json');
-const commonjs = require('rollup-plugin-commonjs');
-const nodeBuiltins = require('rollup-plugin-node-builtins');
-const nodeResolve = require('rollup-plugin-node-resolve');
-const babel = require('rollup-plugin-babel');
+// Entry point of the library
+const input = './lib/bson.js';
 
-const input = 'lib/bson.js';
-const plugins = [
-  nodeResolve(),
-  commonjs(),
-  nodeBuiltins(),
-  babel({
-    externalHelpers: true,
-    presets: [
-      [
-        '@babel/env',
-        {
-          modules: false
-        }
-      ]
-    ]
-  })
-];
-
-const browserPlugins = [
-  nodeResolve({
-    browser: true,
-    preferBuiltins: false
-  }),
-  commonjs(),
-  nodeBuiltins(),
-  babel({
-    externalHelpers: true,
-    presets: [
-      [
-        '@babel/env',
-        {
-          modules: false
-        }
-      ]
-    ]
-  })
-];
-
-const external = Object.keys(pkg.dependencies || {});
-const defaultName = 'BSON';
-
-module.exports = [
-  {
-    input,
-    output: {
-      file: 'dist/bson.esm.js',
-      format: 'es',
-      name: defaultName,
-      exports: 'named'
-    },
-    plugins,
-    external
-  },
+export default [
   {
     input,
-    output: {
-      file: 'dist/bson.browser.umd.js',
-      format: 'umd',
-      name: defaultName,
-      exports: 'named',
-      globals: {
-        buffer: 'Buffer'
+    output: [
+      /* Browser ESM bundle */
+      {
+        file: 'dist/bson.browser.esm.js',
+        format: 'es'
+      },
+      /* Browser UMD bundle */
+      {
+        file: 'dist/bson.browser.umd.js',
+        format: 'umd',
+        name: 'BSON'
+      },
+      /* Browser IIFE bundle */
+      {
+        file: 'dist/bson.bundle.js',
+        format: 'iife',
+        name: 'BSON'
       }
-    },
-    plugins: browserPlugins,
-    external
-  },
-  {
-    input,
-    output: {
-      file: 'dist/bson.browser.esm.js',
-      format: 'es',
-      name: defaultName,
-      exports: 'named'
-    },
-    plugins: browserPlugins,
-    external
+    ],
+
+    plugins: [
+      nodeResolve({ preferBuiltins: false }),
```

Comment:
in both cases you are using `preferBuiltins: false`, and there is a removed `nodeBuiltins()` plugin. Can you explain what's going on here?

Reply:
In the `dist/bson.esm.js` case it is a redundant setting since we have  `external: ['buffer']` it won't bundle it the locally installed one nor the node one. This bundle is the odd one out kinda, and whether or not we need nodePolyfills (replacement for nodeBuiltins)  depends on who is consuming it. If the target is just node then I can just have the commonjs() plugin only I believe.

Just some side info:
I learned on my bundling journey that these appear to be abandoned:
rollup-plugin-node-globals
rollup-plugin-node-builtins
This is a remake of the builtins lib and rollup wants to bring it into their monorepo for plugins but that has yet to happen
rollup-plugin-node-polyfills